src/NServiceBus.Core/BusConfiguration.cs

```diff
@@ -267,11 +267,18 @@ internal InitializableEndpoint Build()
         {
             ForAllTypes<T>(types, t =>
             {
+                if (!HasDefaultConstructor(t))
+                {
+                    throw new ArgumentException($"Unable to create the type '{t.Name}'. Types implementing '{typeof(T).Name}' must have a parameterless (default) constructor.");
```

Comment:
Since this will also throw for an intenal and private constructor should also add that it needs to be public